src/components/types/k8s/cluster-deployment.ts

```diff
@@ -0,0 +1,22 @@
+import { K8sResourceCommon } from '@openshift-console/dynamic-plugin-sdk';
+
+export type ClusterInstallRef = {
+  group: string;
+  kind: string;
+  version: string;
+  name: string;
+};
+
+export type ClusterDeploymentK8sResource = K8sResourceCommon & {
+  spec?: {
+    baseDomain: string;
+    clusterInstallRef: ClusterInstallRef;
+    clusterName: string;
+    platform: {
+      agentBareMetal: {
+        agentSelector: any;
```

Comment:
I wonder if any of those fields is optional. Can you please confirm? Just to be safe ...

Reply:
I made some of the fields optional, let's update the rest in follow ups as needed